contribs/gmf/src/directives/locationchooser.js

```diff
@@ -76,7 +73,14 @@ gmf.LocationchooserController = function($scope) {
    * @type {gmfx.LocationchooserLocation}
    * @export
    */
-  this.selectedLocation = this.locations[0];
+  this.selectedLocation;
+
+  if (!goog.isDef(this.selectedLocation)) {
+    this.selectedLocation = this.locations[0];
+  }
+
+  $scope.$watch('ctrl.selectedLocation', angular.bind(this, this.setLocation));
```

Comment:
Also, I'd recommend using:

```js
$scope.$watch(goog.bind(function() {
  return this.selectedLocation;
}, this), goog.bind(this.setLocation, this));
```

To avoid referring to `ctrl` from the js code.